File: Homework 2/Statistics HW 2 C# TIMER AND RANDOM/Statistics HW 2/Form1.cs
using System.Configuration;
using System.Security.Cryptography;
using static System.Windows.Forms.VisualStyles.VisualStyleElement.Rebar;
using Timer = System.Windows.Forms.Timer;

namespace Statistics_HW_2
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        Random rand = new();
        int timeleft = 30;

        private void button1_Click(object sender, EventArgs e)
        {
            timer1.Interval = 1000;
            timer1.Start();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            timer1.Stop();
        }

        private void button3_Click(object sender, EventArgs e)
        {
            timer1.Stop();
            timeleft = 30;
            timervalue.Text = "" + timeleft;
            label1.Text = "";
        }

       
        private void timer1_Tick(object sender, EventArgs e)
        {
            if (timeleft > 0)
            {
                timeleft -= 1;
                timervalue.Text = "" + timeleft;
                label1.Text = "" + rand.Next(100);
            }
            else
            {
                timer1.Stop();
            }
        }

    }
}

File: Homework 2/Statistics HW 2 VB.NET TIMER AND RANDOM/Statistics HW 2 VB.NET/Form1.vb


Imports System.Windows.Forms.VisualStyles.VisualStyleElement.Header

Public Class Form1

    Shared random As New Random()
    Dim timeleft = 30
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        timer1.Interval = 1000
        timer1.Start()
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Timer1.Stop()
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Timer1.Stop()
        timeleft = 30
        Label1.Text = timeleft.ToString()
        Label2.Text = ""
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        If (timeleft > 0) Then
            timeleft -= 1
            Label1.Text = timeleft.ToString()
            Label2.Text = random.Next(100)
        Else
            Timer1.Stop()
        End If
    End Sub
End Class
